### PR TITLE
Unlisted map view

### PR DESCRIPTION
**Pull request by tairea**
this allows 'non-signed-in-users' to go past PopupGalaxyPreview and now see GalaxyView without sigining in.

'non-signed-in-users' are now required to sign in to get past SolarSystemInfoPanel (Popup) to be able to see SolarSystemView.

this creates a better user experience for new users while also making it easier to share 'unlisted' galaxy maps (you can now share a galaxy map url with a non-signed-in-user.

### CODE REVIEW

File: src/store/solarSystemView.ts [hunk 1]
```diff
@@ -24,6 +24,9 @@ export const useSolarSystemViewStore = defineStore({
     setStudentsView(view: boolean) {
       this.studentsView = view;
     },
+    setTopic(topic: ITopic) {
```
Comment (charsleysa) [inline src/store/solarSystemView.ts:27]: don't do this, the store should be responsible for loading the topic

File: src/views/SolarSystemView.vue [hunk 2]
```diff
@@ -253,6 +251,7 @@ export default {
       "setStartMissionLoading",
     ]),
     ...mapActions(useSolarSystemViewStore, [
+      "setTopic",
```
Comment (charsleysa) [inline src/views/SolarSystemView.vue:254]: not used

**Review by charsleysa — CHANGES_REQUESTED** [review]

File: functions/src/courseManagement.ts [hunk 3]
```diff
@@ -1312,9 +1312,12 @@ async function updateTaskOrderIndexes(
         .collection("tasks")
         .doc(taskOrderIndex.taskId);
 
-      await taskRef.update({
-        orderIndex: taskOrderIndex.orderIndex,
-      });
+      await taskRef.set(
```
[the remark this answers is not in the document]
Comment (tairea) [reply functions/src/courseManagement.ts:1315]: .update() was removing fields that weren't being updated, such as .taskStatus it was also changing timestamp formats. was a headache. set with merge: true did what we wanted 🤷‍♂️

File: src/views/SolarSystemView.vue (hunk 2 above)
Comment (tairea) [reply src/views/SolarSystemView.vue:254]: removed

File: src/store/solarSystemView.ts (hunk 1 above)
Comment (tairea) [reply src/store/solarSystemView.ts:27]: removed

File: functions/src/courseManagement.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (charsleysa) [reply functions/src/courseManagement.ts:1315]: according to the firestore documentation, it shouldn't be doing that
https://cloud.google.com/firestore/docs/samples/firestore-data-set-field-async

this change could be masking a more serious bug, caution should be taken

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**